test/functional/tests/cli/test_cli_start_stop.py

```diff
@@ -15,76 +15,193 @@
         cache_device.create_partitions([Size(500, Unit.MebiByte)])
         cache_device = cache_device.partitions[0]
 
-    with TestRun.LOGGER.step("Start cache"):
-        casadm.start_cache(cache_device, shortcut=shortcut, force=True)
+    with TestRun.step("Start the cache."):
+        cache = casadm.start_cache(cache_device, shortcut=shortcut, force=True)
 
-    with TestRun.LOGGER.step("Check if cache started successfully"):
+    with TestRun.step("Check if the cache has started successfully."):
         caches = casadm_parser.get_caches()
         if len(caches) != 1:
-            TestRun.LOGGER.error(f"There is wrong caches count found in OS: {len(caches)}")
-        if caches[0].cache_device.system_path != cache_device.system_path:
-            TestRun.LOGGER.error(f"Cache started using wrong device: "
-                                 f"{caches[0].cache_device.system_path}. "
-                                 f"Should be {cache_device.system_path}")
+            TestRun.fail(f"There is wrong number of caches found in the OS: {len(caches)}. "
+                         f"Should be only 1.")
+        if cache.cache_device.system_path != cache_device.system_path:
+            TestRun.fail(f"The cache started using wrong device: {cache.cache_device.system_path}."
+                         f"\nShould use {cache_device.system_path}.")
 
-    with TestRun.LOGGER.step("Stop cache"):
-        casadm.stop_cache(cache_id=caches[0].cache_id, shortcut=shortcut)
+    with TestRun.step("Stop the cache."):
+        casadm.stop_cache(cache.cache_id, shortcut=shortcut)
 
-    with TestRun.LOGGER.step("Check if cache stopped properly"):
+    with TestRun.step("Check if the cache has stopped properly."):
+        caches = casadm_parser.get_caches()
+        if len(caches) != 0:
+            TestRun.fail(f"There is wrong number of caches found in the OS: {len(caches)}."
+                         f"\nNo cache should be present after stopping the cache.")
         output = casadm.list_caches(shortcut=shortcut)
+        cli_messages.check_stdout_msg(output, cli_messages.no_caches_running)
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.nand, DiskType.optane]))
+@pytest.mark.parametrize("shortcut", [True, False])
+def test_cli_start_stop_custom_id(shortcut):
+    """
+        title: Test for starting cache with custom ID - short and long command version
+        description: |
+          Start the new cache with random ID (from allowed pool) and then stop this cache.
+        pass_criteria:
+          - The cache has successfully started with custom ID
+          - The cache has successfully stopped
+    """
+    with TestRun.step("Prepare the device for the cache."):
+        cache_device = TestRun.disks['cache']
+        cache_device.create_partitions([Size(500, Unit.MebiByte)])
+        cache_device = cache_device.partitions[0]
+
+    with TestRun.step("Start the cache with random ID."):
+        cache_id = randint(1, 16384)
```

Comment:
I'd either assign those numbers to some MIN/MAX_ALLOWED_ID variables or I'd add a comment, to clarify those numbers are not picked at random, but they are significant